[19.0][MIG] server_action_mass_edit: Migration to 19.0

jcadhoc opened the pull request.

legalsylvain committed:
[REF] rename 'mass_editing' module into 'server_action_mass_edit'

legalsylvain committed:
[UPD] french translation

victoralmau committed:
[IMP] server_action_mass_edit: Do not re-use demo and admin users to avoid incompatibilities with data from those users.

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (43 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/tr/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 18.6% (8 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/sl/

hbrunn committed:
[ADD] server_action_mass_edit: add support for o2m fields

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (43 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 58.1% (25 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/it/

gregory-moka committed:
Translated using Weblate (French)

Currently translated at 95.3% (41 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/fr/

jappi00 committed:
Translated using Weblate (German)

Currently translated at 100.0% (43 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/de/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (43 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/it/

Commit:
[FIX] server_action_mass_edit: 'Add' readded to many2many

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (43 of 43 strings)

Translation: server-ux-16.0/server-ux-16.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-server_action_mass_edit/sv/

rov-adhoc committed:
[MIG] server_action_mass_edit: Migration to 17.0

CRogos committed:
[IMP] server_action_mass_edit: black, isort, prettier

Commit:
[UPD] Update server_action_mass_edit.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-17.0/server-ux-17.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-server_action_mass_edit/

alan196 committed:
Translated using Weblate (Spanish (Mexico))

Currently translated at 100.0% (44 of 44 strings)

Translation: server-ux-17.0/server-ux-17.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-server_action_mass_edit/es_MX/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: server-ux-17.0/server-ux-17.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-server_action_mass_edit/it/

Vang-NguyenPhu committed:
[FIX] server_action_mess_edit: Application error when discard a x2many field

OCA-git-bot committed:
[BOT] post-merge updates

cvinh committed:
[IMP] Improve documentation during OCA Days 2024 Workshop thanks to @Tumbaoju

OCA-git-bot committed:
[BOT] post-merge updates

trisdoan committed:
[MIG] server_action_mass_edit: Migration to 18.0

add pre-init-hook

trisdoan committed:
[IMP] server_action_mass_edit: pre-commit auto fixes

trisdoan committed:
[ADD] server_action_mass_edit_onchange: extension to support playing onchange

Commit:
[UPD] Update server_action_mass_edit.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-18.0/server-ux-18.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-18-0/server-ux-18-0-server_action_mass_edit/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: server-ux-18.0/server-ux-18.0-server_action_mass_edit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-18-0/server-ux-18-0-server_action_mass_edit/it/

OCA-git-bot committed:
[BOT] post-merge updates

adavid-tsc committed:
[FIX] server_action_mass_edit: add onchange method for selection__ temp fields removals

OCA-git-bot committed:
[BOT] post-merge updates

jcadhoc committed:
[IMP] : pre-commit auto fixes

Reyes4711-S73 reviewed (approved):
LGTM

Reyes4711-S73 commented:
@jcadhoc Functional review Ok. It's working fine.
Please, can you solve pre-commit and tests errors?

em230418 commented:
Should we wait for finishing migration in this PR or I can  start porting this module (after customer agrees) myself?

jcadhoc committed:
[FIX] server_action_mass_edit: Properly handle read() with None/empty fields

When fields=None or fields=[] is passed to read(), explicitly provide
the list of real model fields from _fields.keys() to avoid Odoo trying
to read dynamic fields that may be in the record cache but not in _fields.

rven commented:
@jcadhoc Can you please follow the guidelines when migrating a module?

[[IMP] : pre-commit auto fixes](https://github.com/OCA/server-ux/pull/1182/changes/429c564e1642257d19f382f6de95a4249b687eb1) -> contains also the migration stuff.

Follow the Full Process documented in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-19.0

rven commented on server_action_mass_edit/__manifest__.py:
```diff
@@ -0,0 +1,32 @@
+        ]
+    },
+    "demo": ["demo/mass_editing.xml"],
+    "external_dependencies": {"python": ["openupgradelib"]},
```
Is this external_dependencies needed? No migrations scripts are in place at the moment.

isturiz replied:
Is for the import in the file [hooks.py](https://github.com/OCA/server-ux/pull/1174/changes#diff-76d76277700a9694e3c019a6b9ab47aed8c6b0fe49c8409ac8ecd4ae772dc3faR3)

However, is this hook actually necessary? I've checked, and this hook was included in v18, but now, in v19, is it still required?

I think so, but I'm not sure about this.

If someone migrates from 17 to 19, the hook can be required.

eduezerouali-tecnativa replied:
Not needed it at all.  You have the migrations script here https://github.com/OCA/server-ux/blob/16.0/server_action_mass_edit/migrations/16.0.2.0.0/pre-migration.py.

rven commented on server_action_mass_edit/demo/mass_editing.xml:
```diff
@@ -0,0 +1,76 @@
+        <field name="server_action_id" ref="mass_editing_user" />
+        <field name="field_id" ref="base.field_res_users__title" />
+    </record>
+    <record id="mass_editing_user_line_9" model="ir.actions.server.mass.edit.line">
+        <field name="server_action_id" ref="mass_editing_user" />
+        <field name="field_id" ref="base.field_res_users__company_type" />
+    </record>
+    <record id="mass_editing_user_line_10" model="ir.actions.server.mass.edit.line">
+        <field name="server_action_id" ref="mass_editing_user" />
+        <field name="field_id" ref="base.field_res_users__image_1920" />
+        <field name="widget_option">image</field>
+    </record>
+    <record id="mass_editing_user_line_11" model="ir.actions.server.mass.edit.line">
+        <field name="server_action_id" ref="mass_editing_user" />
+        <field name="field_id" ref="base.field_res_users__bank_ids" />
+    </record>
```
ValueError: External ID not found in the system: base.field_res_users__title

```suggestion
    <record id="mass_editing_user_line_8" model="ir.actions.server.mass.edit.line">
        <field name="server_action_id" ref="mass_editing_user" />
        <field name="field_id" ref="base.field_res_users__company_type" />
    </record>
    <record id="mass_editing_user_line_9" model="ir.actions.server.mass.edit.line">
        <field name="server_action_id" ref="mass_editing_user" />
        <field name="field_id" ref="base.field_res_users__image_1920" />
        <field name="widget_option">image</field>
    </record>
    <record id="mass_editing_user_line_10" model="ir.actions.server.mass.edit.line">
        <field name="server_action_id" ref="mass_editing_user" />
        <field name="field_id" ref="base.field_res_users__bank_ids" />
    </record>
```

jcadhoc committed:
[FIX] server_action_mass_edit: Fix demo data for Odoo 19 compatibility

- Remove references to fields that don't exist in Odoo 19:
  * category_id (removed in recent versions)
  * comment (doesn't exist in res.users)
  * bank_ids (not directly accessible from res.users)
- Replace with valid fields: company_id, active, login
- Fix line sequence (was missing line_8)

jcadhoc committed:
[FIX] server_action_mass_edit: Use only direct res.users fields in demo

Replace partner fields (country_id, is_company, company_type) with
actual res.users fields (partner_id, signature, tz) to ensure
external IDs exist in base module.

ahmed-aly-aut commented:
any update on this issue

mmircoli-nexapp commented:
is ready to review?

lef-adhoc committed:
[IMP] server_action_mass_edit: custom domain for relational fields

Adds a Domain column to mass edit lines. When set on a relational field (many2one, many2many, one2many), the domain filters the available options in the wizard. Mutually exclusive with Apply Domain.

isturiz commented:
LGTM on runboat!

baf-adhoc reviewed (approved):
LGTM

CRogos commented:
@rven could you update your review?

CRogos reviewed (changes requested):
@jcadhoc could you check the commit messages?
They should be only 2 new commits:
[IMP] server_action_mass_edit: pre-commit auto fixes
[MIG] server_action_mass_edit: Migration to 19.0

<img width="1270" height="1011" alt="Image" src="https://github.com/user-attachments/assets/97f98bde-b1d0-435e-9d5c-e0663b95bde8" />

There are also samples marked yellow of commit which should be squashed. (the marked are only samples, there are more) 
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

andreampiovesana reviewed (approved):
merge?

ahaly-cloudunify commented:
I tried to install this module and has to add the external library `openupgradelib` to my python dependencies and when installing the module I got the following error

```
The operation cannot be completed: Missing required value for the field 'Parent' (parent_id).
Model: 'Model Inheritance Tree' (ir.model.inherit)
- create/update: a mandatory field is not set
- delete: another model requires the record being deleted, you can archive it instead
```

CRogos commented on server_action_mass_edit/hooks.py:
```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 Camptocamp (<https://www.camptocamp.com>)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+from openupgradelib import openupgrade
+
+
+def pre_init_hook(env):
+    if openupgrade.table_exists(env.cr, "mass_editing_line"):
+        openupgrade.rename_models(
+            env.cr, [("mass.editing.line", "ir.actions.server.mass.edit.line")]
+        )
+        openupgrade.rename_tables(
+            env.cr, [("mass_editing_line", "ir_actions_server_mass_edit_line")]
+        )
+
+        modules = [("mass_editing", "server_action_mass_edit")]
+        openupgrade.update_module_names(env.cr, modules, merge_modules=True)
```
I haven't tested it, but the issue of @ahaly-cloudunify is most likely here. Shouldn't this be an migration script and not an init script?

eduezerouali-tecnativa replied:
We do have the script here https://github.com/OCA/server-ux/blob/16.0/server_action_mass_edit/migrations/16.0.2.0.0/pre-migration.py 
Not really sure why they add it.

@jcadhoc This is not longer needed. Please include this https://github.com/OCA/server-ux/pull/1272

eduezerouali-tecnativa reviewed (commented):
cc @Tecnativa TT58710

@jcadhoc could you attend @CRogos comment please.

peluko00 commented:
Any update @jcadhoc?

mmircoli-nexapp commented:
Any update @jcadhoc ?

JanQubiQ commented:
Merge?

legalsylvain commented:
@CRogos could you update your review ?

eduezerouali-tecnativa commented:
@legalsylvain this https://github.com/OCA/server-ux/pull/1272 should be added. Anyways it can be merge and fix later.

CRogos commented:
There was no code change since the review. The history is not squashed. 
Do you think using a hook instead of a migration script is the right way?  (or even remove)
I haven't tested the module/installation/upgrade yet, so I cannot tell if the code is working.

> Anyways it can be merge and fix later.

... or it could be done in a different PR correct in the first place ;)

pedrobaeza commented:
Superseded by #1301
